src/versions.ts

```diff
@@ -62,6 +88,33 @@ function hasVersion(val: unknown): val is { version: unknown } {
   return typeof val === 'object' && val !== null && 'version' in val;
 }
 
+function isReleaseInfo(val: unknown): val is ReleaseInfo {
+  return (
+    typeof val === 'object' &&
+    val !== null &&
+    'version' in val &&
+    typeof (val as { version: unknown }).version === 'string' &&
```

Comment:
Any ways in which we can avoid type casting here?

Reply:
Good question to ask. 🙂

I don't believe so in this case. This is a user-defined type guard using a type predicate and [the TypeScript documentation shows using type casting](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates) in their example. It appears to be a limitation of TypeScript that it can't determine that it's safe to access those properties after using the `in` operator to check that they exist.

Actually, after writing the above, I wanted to double check my understanding using [TS Playground](https://www.typescriptlang.org/play) and to my surprise it didn't have an error when using version 4.9.5. [It looks like in TypeScript 4.9 they fixed this limitation with the `in` operator](https://devblogs.microsoft.com/typescript/announcing-typescript-4-9/#unlisted-property-narrowing-with-the-in-operator).

So, yes, it looks like we can if we update to TypeScript 4.9+. I can do a follow-up PR that updates the TypeScript version separately since there might be other changes related to that so best not to do it in this PR.